### PR TITLE
Pair with negative fee is not allowed

### PR DESCRIPTION
**Pull request by amandeeprehal-lab49**
Pair with negative fee is not allowed

### CODE REVIEW

File: contracts/Pair.sol [hunk 1]
```diff
@@ -48,6 +48,7 @@ contract Pair is IPair, Initializable, TokenOperations {
         address _treasury,
         int256 _fee
     ) public override initializer {
+        require(_fee >= 0, "Pair: Fee should be greater than zero.");
```
Comment (shivamgarglab49) [inline contracts/Pair.sol:51]: if we expecting fee should be greater than zero, then condition should be > 0 only ?

File: test/pair-test.ts [hunk 2]
```diff
@@ -157,41 +157,128 @@ describe("All Tests", function () {
...
-      precision.mul(50),
-      precision.mul(50)
-    );
-    await tx.wait();
-    const tokenQty = await pair.getPairQty();
-    expect(tokenQty[0]).to.be.equals(precision.mul(50));
-    expect(tokenQty[1]).to.be.equals(precision.mul(50));
-  });
+  describe("HBAR pool test cases", async function () {
+    it("Add liquidity to the pool by adding 50 units of HBAR and 50 units of token B  ", async function () {
```
Comment (shivamgarglab49) [inline test/pair-test.ts:161]: I think we need to pass HBars in the same way like :-

{
          value: ethers.utils.parseEther("0.0000000050"),
        }

File: test/pair-test.ts [hunk 3]
```diff
@@ -980,6 +1012,25 @@ describe("All Tests", function () {
     expect(tokenQty[1]).to.be.equals(precision.mul(50));
   });
 
+  it("When user try to create pair with negative fee it should fail ", async () => {
+    const Pair = await ethers.getContractFactory("Pair");
```
Comment (shivamgarglab49) [inline test/pair-test.ts:1016]: I think, we can call initialize method on Pair object also. (May be not required  { unsafeAllow: ["delegatecall"] } with that.)

File: test/pair-test.ts (hunk 2 above)
Comment (amandeeprehal-lab49) [reply test/pair-test.ts:161]: This test was redundant, I have removed it we have existing test covering functionality. `Add liquidity for HBAR`

File: test/pair-test.ts (hunk 3 above)
Comment (amandeeprehal-lab49) [reply test/pair-test.ts:1006]: As discussed not required now

**Review by shivamgarglab49 — APPROVED** [review]

**Pull request merged**